packages/Meshfree/src/DTK_SplineOperator_def.hpp

```diff
@@ -0,0 +1,300 @@
+        Thyra::add<SC>( thyra_PpM, thyra_P_T );
+
+    // Create parameters for stratimikos to setup the inverse operator.
+    auto d_stratimikos_list = Teuchos::parameterList( "Stratimikos" );
+    d_stratimikos_list->set( "Linear Solver Type", "Belos" );
+    d_stratimikos_list->set( "Preconditioner Type", "None" );
+    auto &linear_solver_types_list =
+        d_stratimikos_list->sublist( "Linear Solver Types" );
+    auto &belos_list = linear_solver_types_list.sublist( "Belos" );
+    belos_list.set( "Solver Type", "Pseudo Block GMRES" );
+    auto &solver_types_list = belos_list.sublist( "Solver Types" );
+    auto &gmres_list = solver_types_list.sublist( "Pseudo Block GMRES" );
+    gmres_list.set( "Convergence Tolerance", 1e-10 );
+    gmres_list.set( "Verbosity",
+                    Belos::Errors + Belos::Warnings + Belos::TimingDetails +
+                        Belos::FinalSummary + Belos::StatusTestDetails );
+    gmres_list.set( "Output Frequency", 1 );
+
+    // Create the inverse of the composite operator C.
+    Stratimikos::DefaultLinearSolverBuilder builder;
+    builder.setParameterList( d_stratimikos_list );
+    Teuchos::RCP<Thyra::LinearOpWithSolveFactoryBase<SC>> factory =
+        Thyra::createLinearSolveStrategy( builder );
+    Teuchos::RCP<const Thyra::LinearOpBase<SC>> thyra_C_inv =
+        Thyra::inverse<SC>( *factory, thyra_C );
+
+    // Create the composite operator B = (Q + N);
+    Teuchos::RCP<const Thyra::LinearOpBase<SC>> thyra_B =
+        Thyra::add<SC>( thyra_Q, thyra_N );
+
+    // Create the coupling matrix A = (B * C^-1 * S).
+    _thyra_operator = Thyra::multiply<SC>( thyra_B, thyra_C_inv, thyra_S );
+    DTK_ENSURE( Teuchos::nonnull( _thyra_operator ) );
+
+    _source = Teuchos::rcp( new Vector( S->getDomainMap(), 1 ) );
+    _destination = Teuchos::rcp( new Vector( N->getRangeMap(), 1 ) );
+    _thyra_X = Thyra::createMultiVector<SC>( _source );
+    _thyra_Y = Thyra::createMultiVector<SC>( _destination );
+}
+
+template <typename DeviceType, typename CompactlySupportedRadialBasisFunction,
+          typename PolynomialBasis>
+void SplineOperator<DeviceType, CompactlySupportedRadialBasisFunction,
+                    PolynomialBasis>::
+    apply( Kokkos::View<double const *, DeviceType> source_values,
+           Kokkos::View<double *, DeviceType> target_values ) const
+{
+    // Precondition: check that the source and the target are properly sized
+    DTK_REQUIRE( source_values.extent( 0 ) ==
+                 S->getDomainMap()->getNodeNumElements() );
+    DTK_REQUIRE( target_values.extent( 0 ) ==
+                 N->getRangeMap()->getNodeNumElements() );
+
+    auto domain_map = S->getDomainMap();
+    auto range_map = N->getRangeMap();
+
+    auto source = Teuchos::rcp( new Vector( domain_map, 1 ) );
+    auto destination = Teuchos::rcp( new Vector( range_map, 1 ) );
+
+    Kokkos::deep_copy(
+        Kokkos::subview( _source->getLocalViewDevice(), Kokkos::ALL, 0 ),
```

Comment:
Why do you need the subview?

Reply:
Yes, `getLocalView*` in Tpetra MultiVectors retuns 2D view.